sympy/concrete/summations.py

```diff
@@ -253,6 +257,202 @@ def _eval_simplify(self, ratio, measure):
+                        Sum(sequence_term, (sym, S.NegativeInfinity, 0)).is_convergent()
+            sequence_term = simplify(sequence_term.xreplace({sym: -sym}))
+            lower_limit = -upper_limit
+            upper_limit = S.Infinity
+
+        interval = Interval(lower_limit, upper_limit)
+
+        # Piecewise function handle
+        if sequence_term.is_Piecewise:
+            for func_cond in sequence_term.args:
+                if func_cond[1].func is Ge or func_cond[1].func is Gt or func_cond[1] == True:
+                    return Sum(func_cond[0], (sym, lower_limit, upper_limit)).is_convergent()
+            return True
+
+        ###  -------- Divergence test ----------- ###
+        try:
+            if limit(sequence_term, sym, upper_limit) != S.Zero:
+                return False
+        except NotImplementedError:
+            pass
+
+        order = O(sequence_term, (sym, S.Infinity))
+
+        ### --------- p-series test (1/n**p) ---------- ###
+        p1_series_test = order.expr.match(sym**p)
+        if p1_series_test is not None:
+            if p1_series_test[p] < -1:
+                return True
+            if p1_series_test[p] > -1:
+                return False
+
+        p2_series_test = order.expr.match((1/sym)**p)
+        if p2_series_test is not None:
+            if p2_series_test[p] > 1:
+                return True
+            if p2_series_test[p] < 1:
+                return False
+
+        ### ----------- root test ---------------- ###
+        lim = Limit(abs(sequence_term)**(1/sym), sym, S.Infinity)
+        lim_evaluated = lim.doit()
+        if lim_evaluated.is_number:
+            if lim_evaluated < 1:
+                return True
+            if lim_evaluated > 1:
+                return False
+
+        ### ------------- alternating series test ----------- ###
+        d = symbols('d', cls=Dummy)
+        dict_val = sequence_term.match((-1)**p*q)
+        if dict_val[p] == sym or dict_val[p].has(sym):
+            if is_decreasing(dict_val[q], interval):
+                return True
+            return False
+
+        ### ------------- comparison test ------------- ###
+        # (1/log(n)**p) comparison
+        log_test = order.expr.match(1/(log(sym)**p))
+        if log_test is not None:
+            return False
+
+        # (1/(n*log(n)**p)) comparison
+        log_n_test = order.expr.match(1/(sym*(log(sym))**p))
+        if log_n_test is not None:
+            if log_n_test[p] > 1:
+                return True
+            return False
+
+        # (1/(n*log(n)*log(log(n))*p)) comparison
+        log_log_n_test = order.expr.match(1/(sym*(log(sym)*log(log(sym))**p)))
+        if log_log_n_test is not None:
+            if log_log_n_test[p] > 1:
+                return True
+            return False
+
+        if order.expr == sequence_term:
+            pass
+        else:
+            return Sum(order.expr, (sym, lower_limit, upper_limit)).is_convergent()
```

Comment:
No, that's un-necessary. I forgot to remove this code. Very thank you for
this

On Mon, Oct 5, 2015 at 1:43 PM, Kalevi Suominen <notifications@github.com>
wrote:

> In sympy/concrete/summations.py
> <https://github.com/sympy/sympy/pull/9906#discussion_r41119236>:
>
> > +        if log_n_test is not None:
> > +            if log_n_test[p] > 1:
> > +                return True
> > +            return False
> > +
> > +        # (1/(n*log(n)*log(log(n))*p)) comparison
> > +        log_log_n_test = order.expr.match(1/(sym*(log(sym)*log(log(sym))**p)))
> > +        if log_log_n_test is not None:
> > +            if log_log_n_test[p] > 1:
> > +                return True
> > +            return False
> > +
> > +        if order.expr == sequence_term:
> > +            pass
> > +        else:
> > +            return Sum(order.expr, (sym, lower_limit, upper_limit)).is_convergent()
>
> Is is necessary to repeat the above steps with order.expr? What would be
> different if we just replace the sequence_term with order.expr and
> continue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9906/files#r41119236>.
>